sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2557,6 +2557,10 @@ class AstBuilder extends DataTypeAstBuilder
   }
 
   override def visitCollateClause(ctx: CollateClauseContext): String = withOrigin(ctx) {
```

Comment:
will the collate parser eventually invoke `CollateExpressionBuilder`?

Reply:
No, for statements like: SELECT 'aaa' COLLATE UNICODE_LTRIM_CI it won't happen.